File: 5.implementation/VT/redsun/src/main/webapp/static/js/services/QualitiesService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('qualitieService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(qualitie) {
			var urlServer = $rootScope.makePostURL("/qualitie/insert");
			return $http.post(urlServer, qualitie);
		}
		
		// Update.
		, update: function(qualitieId, qualitie) {
			var urlServer = $rootScope.makePostURL("/qualitie/update/" + qualitieId);
			return $http.put(urlServer, qualitie);
		}
		
		// Delete.
		, delete: function(qualitieId) {
			var urlServer = $rootScope.makePostURL("/qualitie/delete/" + qualitieId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(qualitieId) {
			var url = $rootScope.makeGetURL("/qualitie/getbyid/" + qualitieId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, qualitie) {
			var url = $rootScope.makePostURL("/qualitie/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: qualitie });
		}
		
	}// return.
});

File: 5.implementation/VT/redsun-service/src/main/java/com/redsun/service/service/QualitiesService.java
package com.redsun.service.service;

import com.redsun.service.entities.Qualities;
import com.redsun.service.entities.Result;

/**
 * Qualities Service interface
 */
public interface QualitiesService {
	
    // Insert.
	Result insert(final Qualities qualities) ;

    // Update.
	Result update(final Qualities qualities) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listQualitiesForPageAndFilter(final int itemsPerPage, final int pageNo, final Qualities qualities);

}
